include/ginkgo/core/base/mtx_io.hpp

```diff
@@ -61,6 +61,46 @@ template <typename ValueType = default_precision, typename IndexType = int32>
 matrix_data<ValueType, IndexType> read_raw(std::istream& is);
 
 
+/**
+ * Reads a matrix stored in Ginkgo's binary matrix format from an input stream.
+ * Note that this format depends on the processor's endianness,
+ * so files from a big endian processor can't be read from a little endian
+ * processor and vice-versa.
+ *
+ * @tparam ValueType  type of matrix values
+ * @tparam IndexType  type of matrix indexes
+ *
+ * @param is  input stream from which to read the data
+ *
+ * @return A matrix_data structure containing the matrix. The nonzero elements
+ *         are sorted in lexicographic order of their (row, colum) indexes.
```

Comment:
Your implementation of `read_binary_raw` does not actually sort the matrix data.